app/src/main/java/com/willowtree/vocable/settings/EditCategoriesViewModel.kt

```diff
@@ -25,6 +29,17 @@ class EditCategoriesViewModel(
     private val liveAddRemoveCategoryList = MutableLiveData<List<Category>>()
     val addRemoveCategoryList: LiveData<List<Category>> = liveAddRemoveCategoryList
 
+    val categoryPages = categoriesUseCase.categories().map { categories ->
+        val pageSize = 8
+        val pageCount = ceil(categories.size / pageSize.toFloat()).toInt()
+
+        (0 until pageCount).map {
+            val start = it * pageSize
+            val end = min((it + 1) * pageSize, categories.size)
+            EditCategoriesPage(categories.subList(start, end))
+        }
+    }
+
```

Comment:
seems like this could be a good case for  using `List.chunked(n)` 👀 

```
val categoryPages = categoriesUseCase.categories().map { categories ->
    val pageSize = 8
    categories.chunked(pageSize) { 
        EditCategoriesPage(it)
    }
}
```

Reply:
I knew there was a better API for this! I couldn't remember what it was. You've saved the day, will definitely implement that